src/main/java/org/apache/flink/benchmark/full/OffheapInputWrapper.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.flink.benchmark.full;
+
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.MemorySegment;
+import org.apache.flink.core.memory.MemorySegmentFactory;
+import org.apache.flink.runtime.io.network.api.serialization.SpillingAdaptiveSpanningRecordDeserializer;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.buffer.FreeingBufferRecycler;
+import org.apache.flink.runtime.io.network.buffer.NetworkBuffer;
+
+import java.lang.reflect.Field;
+
+public class OffheapInputWrapper {
+    public SpillingAdaptiveSpanningRecordDeserializer<?> reader;
+    public Buffer buffer;
+    public DataInputView dataInput;
+
+    public OffheapInputWrapper(byte[] initialPayload) throws Exception {
+        reader = new SpillingAdaptiveSpanningRecordDeserializer<>(new String[0]);
+        MemorySegment segment = MemorySegmentFactory.allocateUnpooledOffHeapMemory(initialPayload.length, this);
+        segment.put(0, initialPayload);
+        buffer = new NetworkBuffer(segment, FreeingBufferRecycler.INSTANCE, true, initialPayload.length);
+        Field nonSpanningWrapper = reader.getClass().getDeclaredField("nonSpanningWrapper");
+        nonSpanningWrapper.setAccessible(true);
+        dataInput = (DataInputView) nonSpanningWrapper.get(reader);
+    }
```

Comment:
There are a couple of issues here.

First, this relays on a non public Flink's API (`SpillingAdaptiveSpanningRecordDeserializer`), which means it can change at any point of time and brake the benchmark builds.

Secondly, this is using reflections to access a private field of an `Internal` class, which makes this benchmark even more fragile (and could cause compilation [issues like this recent one](https://issues.apache.org/jira/browse/FLINK-15199)).

Is there a way to avoid using internal classes? If there is really a no way, then it could be fixed similarly how https://github.com/dataArtisans/flink-benchmarks/pull/40 is refactoring the 15199 (take a look at `org.apache.flink.contrib.streaming.state.benchmark.StateBackendBenchmarkUtils` class).

Here in this case, the `OffheapInputWrapper` class should be defined and tested in `flink` repository, with some comment/annotation that it's being used by micro benchmarks. + code should be adjusted that the reflection is no longer needed.